File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //prommt user for an input
    string text = get_string("Text: ");

    //calculate number of letters in text
    int letters = count_letters(text);

    //calculate number of words
    int words = count_words(text);

    //calculate sentences
    int sentences = count_sentences(text);

    //calculate Coleman-Liau index
    float index = 0.0588 * ((float) letters * 100 / words) - 0.296 * ((float) sentences * 100 / words) - 15.8;
    index = round(index);

    //print result
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }
}

int count_letters(string text)
{
    int counter = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            counter++;
        }
    }
    return counter;
}

int count_words(string text)
{
    int counter = 1;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (text[i] == ' ')
        {
            counter++;
        }
    }
    return counter;
}

int count_sentences(string text)
{
    int counter = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            counter++;
        }
    }
    return counter;
}

File: pset6/pset6/readability/readability.py
# get string input from a user
text = input("Text: ")

# count number of letters, words and sentences
letters = sum(c.isalpha() for c in text)
words = len(text.split())
sentences = 0
for c in text:
    if c in [".", "!", "?"]:
        sentences += 1

# letters and sentences per 100 words
letters_100 = letters * 100 / words
sentences_100 = sentences * 100 / words

# calculate Coleman-Liau index
index = round(0.0588 * letters_100 - 0.296 * sentences_100 - 15.8)

# print results
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
